### PR TITLE
[patch] Updates to DB2U install task

### PR DESCRIPTION
**Pull request by kvijai82**
This PR proposes 2 minor updates to the DB2 install task

- The current code assumes that DB2 operator package will always come from the ibm-operator-catalog.  This is not a good assumption if the user is doing a CASE based install.  Use the common create_subscription task which queries the packagemanifests to find the correct source to use instead.

- The current code will execute the `norootsquash` role in all cases today.  This is only required when the user is running on ROKS and is using the `ibmc-file-gold-gid` storageclass.  Running this all the time requires the users to run an additional role to mirror images that are not required in a lot of scenarios.  Updated to conditionally run it based on storageclass the user is using.

### CODE REVIEW

**Comment by kvijai82** [comment]
Output from run with these updates:
```
TASK [ibm.mas_devops.db2 : create_subscription : Set the subscription information] ************************************************************************************************************************
ok: [localhost]

TASK [ibm.mas_devops.db2 : create_subscription: Use default channel if none is provided] ******************************************************************************************************************
skipping: [localhost]

TASK [ibm.mas_devops.db2 : create_subscription: Check if operator group is present in katamari namespace] *************************************************************************************************
ok: [localhost]

TASK [ibm.mas_devops.db2 : create_subscription: Debug settings] *******************************************************************************************************************************************
ok: [localhost] => {
    "msg": [
        "Package Name ............................ db2u-operator",
        "Namespace ............................... katamari",
        "OperatorGroup Exists .................... True",
        "Catalog Source .......................... ibm-db2uoperator-catalog",
        "Catalog Source Namespace ................ openshift-marketplace",
        "Default Channel ......................... v110509.0",
        "Channel ................................. v110509.0",
        "Configuration ........................... <undefined>"
    ]
}

TASK [ibm.mas_devops.db2 : create_subscription: Create db2u-operator subscription in katamari] ************************************************************************************************************
changed: [localhost]

TASK [ibm.mas_devops.db2 : create_subscription : Lookup InstallPlan] **************************************************************************************************************************************
ok: [localhost]
```

**Comment by kvijai82** [comment]
The subscription also got created successfully:
```
[root@aiops-airgap-reg install]# oc get subs db2u-operator -o yaml
apiVersion: operators.coreos.com/v1alpha1
kind: Subscription
metadata:
  creationTimestamp: "2024-02-23T19:10:22Z"
  generation: 1
  labels:
    operators.coreos.com/db2u-operator.katamari: ""
  name: db2u-operator
  namespace: katamari
  resourceVersion: "3111868"
  uid: 014c69da-5330-4073-aa81-2bd1a72a57d3
spec:
  channel: v110509.0
  installPlanApproval: Automatic
  name: db2u-operator
  source: ibm-db2uoperator-catalog
  sourceNamespace: openshift-marketplace
status:
  catalogHealth:
  - catalogSourceRef:
      apiVersion: operators.coreos.com/v1alpha1
      kind: CatalogSource
      name: aiops-analytics-operator-catalog
      namespace: openshift-marketplace
      resourceVersion: "3017732"
      uid: 30396bdc-2bff-4153-adb8-89d1090ff2bc
    healthy: true
    lastUpdated: "2024-02-23T19:10:22Z"
  - catalogSourceRef:
      apiVersion: operators.coreos.com/v1alpha1
      kind: CatalogSource
      name: aiops-lifecycle-operator-catalog
      namespace: openshift-marketplace
      resourceVersion: "3109523"
      uid: 53c97b99-601d-4884-92dd-3337c6a7b533
    healthy: true
    lastUpdated: "2024-02-23T19:10:22Z"
  - catalogSourceRef:
      apiVersion: operators.coreos.com/v1alpha1
      kind: CatalogSource
      name: certified-operators
      namespace: openshift-marketplace
      resourceVersion: "3102329"
      uid: 2280213e-28e5-4e42-a3c7-2e4dcf38ad3b
    healthy: true
    lastUpdated: "2024-02-23T19:10:22Z"
  - catalogSourceRef:
      apiVersion: operators.coreos.com/v1alpha1
      kind: CatalogSource
      name: cloud-native-postgresql-catalog
      namespace: openshift-marketplace
      resourceVersion: "3018466"
      uid: 130d45fc-d159-4229-9362-b6884aa243b7
    healthy: true
    lastUpdated: "2024-02-23T19:10:22Z"
  - catalogSourceRef:
      apiVersion: operators.coreos.com/v1alpha1
      kind: CatalogSource
      name: community-operators
      namespace: openshift-marketplace
      resourceVersion: "3090298"
      uid: 5f8938c7-6e72-4625-9821-9b0c1fa285ad
    healthy: true
    lastUpdated: "2024-02-23T19:10:22Z"
  - catalogSourceRef:
      apiVersion: operators.coreos.com/v1alpha1
      kind: CatalogSource
      name: ibm-aiops-ir-core-operators
      namespace: openshift-marketplace
      resourceVersion: "3109806"
      uid: 233e65df-5275-4d93-8b18-d9f3024473ec
    healthy: true
    lastUpdated: "2024-02-23T19:10:22Z"
  - catalogSourceRef:
      apiVersion: operators.coreos.com/v1alpha1
      kind: CatalogSource
      name: ibm-aiopsedge-case
      namespace: openshift-marketplace
      resourceVersion: "2827937"
      uid: 527ecf0c-b13e-47d3-9d73-a84017a6b180
    healthy: true
    lastUpdated: "2024-02-23T19:10:22Z"
  - catalogSourceRef:
      apiVersion: operators.coreos.com/v1alpha1
      kind: CatalogSource
      name: ibm-asm-catalog
      namespace: openshift-marketplace
      resourceVersion: "3109977"
      uid: fcb37814-d674-4d8b-a855-e3620f197011
    healthy: true
    lastUpdated: "2024-02-23T19:10:22Z"
  - catalogSourceRef:
      apiVersion: operators.coreos.com/v1alpha1
      kind: CatalogSource
      name: ibm-ca-operator-catalog
      namespace: openshift-marketplace
      resourceVersion: "3047705"
      uid: 8e450a35-9663-4077-8e7c-1813f3e718ed
    healthy: true
    lastUpdated: "2024-02-23T19:10:22Z"
  - catalogSourceRef:
      apiVersion: operators.coreos.com/v1alpha1
      kind: CatalogSource
      name: ibm-cert-manager-catalog
      namespace: openshift-marketplace
      resourceVersion: "2827962"
      uid: 12c3a5f0-e76b-4fc8-a107-604e2bc7299f
    healthy: true
    lastUpdated: "2024-02-23T19:10:22Z"
  - catalogSourceRef:
      apiVersion: operators.coreos.com/v1alpha1
      kind: CatalogSource
      name: ibm-cloud-databases-redis-operator-catalog
      namespace: openshift-marketplace
      resourceVersion: "3018356"
      uid: 679ce7d0-248a-460b-bcdf-eb95d5f8f3db
    healthy: true
    lastUpdated: "2024-02-23T19:10:22Z"
  - catalogSourceRef:
      apiVersion: operators.coreos.com/v1alpha1
      kind: CatalogSource
      name: ibm-cp-waiops-catalog
      namespace: openshift-marketplace
      resourceVersion: "2827935"
      uid: 6121daaf-6fc6-4b5b-8965-1bbefbd5524f
    healthy: true
    lastUpdated: "2024-02-23T19:10:22Z"
  - catalogSourceRef:
      apiVersion: operators.coreos.com/v1alpha1
      kind: CatalogSource
      name: ibm-cp-waiops-elastic-catalog
      namespace: openshift-marketplace
      resourceVersion: "2827972"
      uid: 29f9806c-a7d3-4222-967a-668e02ad22d9
    healthy: true
    lastUpdated: "2024-02-23T19:10:22Z"
  - catalogSourceRef:
      apiVersion: operators.coreos.com/v1alpha1
      kind: CatalogSource
      name: ibm-cp-waiops-flink-catalog
      namespace: openshift-marketplace
      resourceVersion: "2827933"
      uid: 9f453633-b813-4de4-8c3d-32d16e11498e
    healthy: true
    lastUpdated: "2024-02-23T19:10:22Z"
  - catalogSourceRef:
      apiVersion: operators.coreos.com/v1alpha1
      kind: CatalogSource
      name: ibm-cpd-ccs-operator-catalog
      namespace: openshift-marketplace
      resourceVersion: "3085232"
      uid: f9470dfd-97b6-49d0-ac3a-ff7e7369e530
    healthy: true
    lastUpdated: "2024-02-23T19:10:22Z"
  - catalogSourceRef:
      apiVersion: operators.coreos.com/v1alpha1
      kind: CatalogSource
      name: ibm-db2oltp-cp4d-operator-catalog
      namespace: openshift-marketplace
      resourceVersion: "3039669"
      uid: 354a9926-46c8-45a3-846c-9a0ac882c6f7
    healthy: true
    lastUpdated: "2024-02-23T19:10:22Z"
  - catalogSourceRef:
      apiVersion: operators.coreos.com/v1alpha1
      kind: CatalogSource
      name: ibm-db2uoperator-catalog
      namespace: openshift-marketplace
      resourceVersion: "3019269"
      uid: df05f733-5d96-4417-98df-5c20c730ee3e
    healthy: true
    lastUpdated: "2024-02-23T19:10:22Z"
  - catalogSourceRef:
      apiVersion: operators.coreos.com/v1alpha1
      kind: CatalogSource
      name: ibm-elasticsearch-catalog
      namespace: openshift-marketplace
      resourceVersion: "3109652"
      uid: 16628cda-adce-4623-be14-6b222046abd3
    healthy: true
    lastUpdated: "2024-02-23T19:10:22Z"
  - catalogSourceRef:
      apiVersion: operators.coreos.com/v1alpha1
      kind: CatalogSource
      name: ibm-kong-management-catalog
      namespace: openshift-marketplace
      resourceVersion: "2827996"
      uid: d28c0f52-83fb-4f35-989f-221f1500c54e
    healthy: true
    lastUpdated: "2024-02-23T19:10:22Z"
  - catalogSourceRef:
      apiVersion: operators.coreos.com/v1alpha1
      kind: CatalogSource
      name: ibm-licensing-catalog
      namespace: openshift-marketplace
      resourceVersion: "2829976"
      uid: b1e8c48f-c5e3-4022-86df-e15c585ad8bb
    healthy: true
    lastUpdated: "2024-02-23T19:10:22Z"
  - catalogSourceRef:
      apiVersion: operators.coreos.com/v1alpha1
      kind: CatalogSource
      name: ibm-redis-cp-operator-catalog
      namespace: openshift-marketplace
      resourceVersion: "2827955"
      uid: 91aa179e-94b1-4c5b-a2e4-4d9bbfe91219
    healthy: true
    lastUpdated: "2024-02-23T19:10:22Z"
  - catalogSourceRef:
      apiVersion: operators.coreos.com/v1alpha1
      kind: CatalogSource
      name: ibm-secure-tunnel-operator-catalog
      namespace: openshift-marketplace
      resourceVersion: "2829984"
      uid: c40e4659-55ce-4894-afbf-b4a0149ca4cb
    healthy: true
    lastUpdated: "2024-02-23T19:10:22Z"
  - catalogSourceRef:
      apiVersion: operators.coreos.com/v1alpha1
      kind: CatalogSource
      name: ibm-watson-ai-manager-catalog
      namespace: openshift-marketplace
      resourceVersion: "3109587"
      uid: 9baa1950-1153-44dd-a8bf-45f7b06fa33b
    healthy: true
    lastUpdated: "2024-02-23T19:10:22Z"
  - catalogSourceRef:
      apiVersion: operators.coreos.com/v1alpha1
      kind: CatalogSource
      name: ibm-watson-aiops-ui-catalog
      namespace: openshift-marketplace
      resourceVersion: "2827939"
      uid: 4a47c8b0-a2c4-480b-82f3-d2262f498578
    healthy: true
    lastUpdated: "2024-02-23T19:10:22Z"
  - catalogSourceRef:
      apiVersion: operators.coreos.com/v1alpha1
      kind: CatalogSource
      name: opencloud-operators
      namespace: openshift-marketplace
      resourceVersion: "2827938"
      uid: 953ae053-c4c5-4526-b350-d02dea110e1e
    healthy: true
    lastUpdated: "2024-02-23T19:10:22Z"
  - catalogSourceRef:
      apiVersion: operators.coreos.com/v1alpha1
      kind: CatalogSource
      name: redhat-marketplace
      namespace: openshift-marketplace
      resourceVersion: "3108408"
      uid: bdfe393e-f193-42b0-b876-32f57b2758e9
    healthy: true
    lastUpdated: "2024-02-23T19:10:22Z"
  - catalogSourceRef:
      apiVersion: operators.coreos.com/v1alpha1
      kind: CatalogSource
      name: redhat-operators
      namespace: openshift-marketplace
      resourceVersion: "3090487"
      uid: 9fe2c1b0-7c55-4876-97f2-797ce650c08f
    healthy: true
    lastUpdated: "2024-02-23T19:10:22Z"
  conditions:
  - lastTransitionTime: "2024-02-23T19:10:22Z"
    message: all available catalogsources are healthy
    reason: AllCatalogSourcesHealthy
    status: "False"
    type: CatalogSourcesUnhealthy
  - status: "False"
    type: BundleUnpacking
  currentCSV: db2u-operator.v110509.0.0
  installPlanGeneration: 10
  installPlanRef:
    apiVersion: operators.coreos.com/v1alpha1
    kind: InstallPlan
    name: install-p4f8x
    namespace: katamari
    resourceVersion: "3110710"
    uid: 04b49695-88d6-487e-b63d-828b420ac051
  installedCSV: db2u-operator.v110509.0.0
  installplan:
    apiVersion: operators.coreos.com/v1alpha1
    kind: InstallPlan
    name: install-p4f8x
    uuid: 04b49695-88d6-487e-b63d-828b420ac051
  lastUpdated: "2024-02-23T19:10:47Z"
  state: AtLatestKnown
```

File: ibm/mas_devops/roles/db2/tasks/install/main.yml [hunk 1]
```diff
@@ -157,11 +157,19 @@
 - name: "Delete old db2 subscription, operand request and csv from {{ ibm_common_services_namespace }}"
   include_tasks: "tasks/delete_db2_operand_request.yml"
 
-- name: "Create Db2 Universal Operator Subscription in {{ db2_namespace }} namespace"
+- name: "Create db2u namespace operatorgroup"
```
Comment (durera) [inline ibm/mas_devops/roles/db2/tasks/install/main.yml:160]: The operatorgroup will be created by `common_tasks/create_subscription.yml` (if one is needed) so we can also remove this (and the associated [templates/db2u_operatorgroup.yml.j2`) ...

**Review by durera — CHANGES_REQUESTED** [review]
One small change ...

**Comment by kvijai82** [comment]
I don't have a ROKS env to test out the `norootsquash` change but I did simulate it with the storageclass name to test in my OCP env to verify it ran the task when that storageclass exists and doesn't run when it doesn't exist.

File: ibm/mas_devops/roles/db2/tasks/install/main.yml (hunk 1 above)
Comment (kvijai82) [reply ibm/mas_devops/roles/db2/tasks/install/main.yml]: Thanks for the feedback.  Removed the call and the template.

Comment (kvijai82) [reply ibm/mas_devops/roles/db2/tasks/install/main.yml]: Updated in the last commit.

File: ibm/mas_devops/roles/db2/tasks/install/main.yml [hunk 2]
```diff
@@ -24,7 +24,10 @@
 # 3. Setup the norootsquash daemonsets for db2u pods to work with NFS backed storage
 # -----------------------------------------------------------------------------
 - include_tasks: tasks/install/setup_norootsquash.yml
-
+  when:
+    - db2_meta_storage_class is defined
+    - db2_data_storage_class is defined
+    - db2_data_storage_class == "ibmc-file-gold-gid" or db2_meta_storage_class == "ibmc-file-gold-gid"
```
Comment (durera) [inline ibm/mas_devops/roles/db2/tasks/install/main.yml:30]: Can we change this to a matching "starts with ibmc-file-" so we catch any use of silver/bronze if it's out there as well.

**Review by durera — CHANGES_REQUESTED** [review]

**Review by durera — APPROVED** [review]

**Pull request merged**